File: src/types/environment.ts
export interface EnvVariable {
  id: string;
  environment_id: string;
  key: string;
  value: string;
  var_type: string;
  enabled: boolean;
}

export interface Environment {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
  variables: EnvVariable[];
}

File: src-tauri/src/models/environment.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Environment {
    pub id: String,
    pub name: String,
    pub created_at: String,
    pub updated_at: String,
    pub variables: Vec<EnvVariable>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct EnvVariable {
    pub id: String,
    pub environment_id: String,
    pub key: String,
    pub value: String,
    pub var_type: String, // 'default', 'secret'
    pub enabled: bool,
}
